arch/arm/soc/silabs_exx32/efr32fg1p/soc.c

```diff
@@ -38,17 +38,21 @@ static const CMU_LFXOInit_TypeDef lfxoInit = CMU_LFXOINIT_DEFAULT;
 static ALWAYS_INLINE void clkInit(void)
 {
 #ifdef CONFIG_CMU_HFCLK_HFXO
-	CMU_HFXOInit(&hfxoInit);
-	CMU_OscillatorEnable(cmuOsc_HFXO, true, true);
-	CMU_ClockSelectSet(cmuClock_HF, cmuSelect_HFXO);
-	CMU_OscillatorEnable(cmuOsc_HFRCO, false, false);
+	if (CMU_ClockSelectGet(cmuClock_HF) != cmuSelect_HFXO) {
```

Comment:
The clocks may have been initialized with a different setting. I think this is not handled here.
So if the bootloader already initializes HFXO, but with different settings, you would get unexpected behaviour here.

Reply:
The job of this part of the code is to enable High Frequency Crystal Oscillator (HFXO) and connect/route it to the CPU. The frequency is defined by the oscillator itself. As such it's not possible to have bootloader initialize it with different setting. If the HFXO clock is already initialized and routed there is nothing to do.

Well, almost. One more thing which this part of the code does is tuning of internal capacitors connected to the crystal to optimize shunt current and peak detection. If someone uses Zephyr based bootloader like MCUboot this will be handled correctly, since tuning will be performed on the first run. If someone uses non Zephyr based bootloader it should also be fine, since traditionally it's a bootloader task to configure low level hardware parameters that the system doesn't have to care about. So it's correct for Zephyr to assume that the tuning has already been performed by the bootloader.

Reply:
OK, for LFXO and HFXO you are right. Tuning parameters, timeouts etc will probably be configured in a similar way in the bootloader and in the main application.